File: oop/base_model.cpp
#ifndef __YEKTANET_BASE_MODEL__
#define __YEKTANET_BASE_MODEL__

#include <bits/stdc++.h>
using namespace std;

class BaseAdvertising {
  private:
    int id;
    int clicks;
    int views;
  public:
    BaseAdvertising(int _id = 0) {
      this->clicks = 0;
      this->views = 0;
      this->id = _id;
    }
    int getClicks() {
      return this->clicks;
    }
    virtual void incClicks() {
      this->clicks++;
    }
    int getViews() {
      return this->views;
    }
    virtual void incViews() {
      this->views++;
    }
    virtual void describeMe() {
      cout << "I am an instance of the baseAdvertising class\n";
    }
};

#endif

File: python/base_model.py
class BaseAdvertising:
    def __init__(self, id = 0):
      self.__clicks = 0
      self.__views = 0
      self.__id = id
    
    def getClicks(self):
      return self.__clicks
    def incClicks(self):
      self.__clicks += 1
    
    def getViews(self):
      return self.__views
    def incViews(self):
      self.__views += 1
    
    def describeMe(self):
      print("I am an instance of the baseAdvertising class\n")